bd-crash-handler/src/lib.rs

```diff
@@ -311,30 +315,32 @@ impl Monitor {
           continue;
         }
 
-        let mut fields = global_state_fields.clone();
-
-        fields.extend(
-          [
-            ("_crash_artifact".into(), contents.into()),
-            (
-              "_crash_reason".into(),
-              crash_reason.unwrap_or_else(|| "unknown".to_string()).into(),
-            ),
-            (
-              "_crash_details".into(),
-              crash_details
-                .unwrap_or_else(|| "unknown".to_string())
-                .into(),
-            ),
-          ]
-          .into_iter(),
-        );
-
-        // TODO(snowp): For now everything in here is a crash, eventually we'll need to be able to
-        // differentiate.
-        // TODO(snowp): Eventually we'll want to upload the report out of band, but for now just
-        // chuck it into a log line.
-        logs.push(CrashLog { fields, timestamp });
+        if let Ok(artifact_id) = self.artifact_client.enqueue_upload(contents).await {
```

Comment:
I don't have a good answer for this, but if this upload eventually gets intent rejected for whatever reason we are going to have to know about that somehow on the backend. So I guess depending on the use case we may have to track rejected state (like if it's duplicated/collapsed or whatever).